lib/operations/command_v2.js

```diff
@@ -15,8 +15,16 @@ class CommandOperationV2 extends OperationBase {
   constructor(parent, options, operationOptions) {
     super(options);
 
+    if (parent.s.logger) {
+      this.logger = parent.s.logger;
+    } else if (parent.s.db && parent.s.db.logger) {
+      this.logger = parent.s.db.logger;
+    }
     this.ns = parent.s.namespace.withCollection('$cmd');
-    this.readPreference = resolveReadPreference(parent, this.options);
+
+    if (this.hasAspect(Aspect.NO_INHERIT_OPTIONS)) parent = {};
```

Comment:
OK, sorry we have a bit of a mess going back and forth on this PR related to `parent` and I'd like to answer this once and for all:

Right now this constructor makes a hard assumption that `parent` exists (see line 18 above, we access `parent.s` without any checks). For the purposes of "inherited properties" here we want either to pass in some class which can have inheritable properites (`MongoClient`/`Db`/`Collection`) or `undefined`, the signature to `ReadPreference.resolve` should have a first parameter `parent?: MongoClient|Db|Collection`, not `parent: MongoClient|Db|Collection|any`

I also think I confused you a bit by suggesting we reuse the `parent` variable for these purposes, let's just go ahead and create a intermediate value to preserve safety and readability:
```
constructor(parent, ..) {

   const propertyProvider = this.hasAspect(Aspect.NO_INHERIT_OPTIONS) ? undefined : parent;
   this.readPreference = ReadPreference.resolve(propertyProvider, this.options);
   ...
}
```